ui/src/app/components/gke-app-configuration-panels/create-gke-app.tsx

```diff
@@ -50,17 +50,75 @@ import { WorkspaceData } from 'app/utils/workspace-data';
 import { CreateGkeAppButton } from './create-gke-app-button';
 import { DisabledCloudComputeProfile } from './disabled-cloud-compute-profile';
 import { OpenGkeAppButton } from './open-gke-app-button';
+
 const defaultIntroText =
   'Your analysis environment consists of an application and compute resources. ' +
   'Your cloud environment is unique to this workspace and not shared with other users.';
 
+const toMachine = (createAppRequest: CreateAppRequest): Machine =>
+  findMachineByName(createAppRequest?.kubernetesRuntimeConfig.machineType);
+
+const toAnalysisConfig = (
+  createAppRequest: CreateAppRequest
+): AnalysisConfig => {
+  const { persistentDiskRequest, kubernetesRuntimeConfig } = createAppRequest;
+  return {
+    machine: toMachine(createAppRequest),
+    diskConfig: {
+      size: persistentDiskRequest.size,
+      detachable: true,
+      detachableType: persistentDiskRequest.diskType,
+      existingDiskName: null,
+    },
+    numNodes: kubernetesRuntimeConfig.numNodes,
+    // defaults
+    computeType: ComputeType.Standard,
+    dataprocConfig: undefined,
+    gpuConfig: undefined,
+    detachedDisk: undefined,
+    autopauseThreshold: undefined,
+  };
+};
+
+const otherApps = (
+  userApps: UserAppEnvironment[] | undefined,
+  thisAppType: AppType
+): UserAppEnvironment[] =>
+  userApps?.filter((app) => app.appType !== thisAppType) ?? [];
+
+const otherMachineTypes = (
+  userApps: UserAppEnvironment[] | undefined,
+  thisAppType: AppType
+): string[] =>
+  Array.from(
+    new Set(
+      otherApps(userApps, thisAppType)
+        .map(
+          (app: UserAppEnvironment) => app.kubernetesRuntimeConfig?.machineType
+        )
+        // filter out undefined values
+        .filter((x) => x) ?? []
+    )
+  );
+
+// if there are other app(s) with a common machine type, return that machine type
+// TODO: what should we do if there is more than one (shouldn't happen) - for now, return undefined
```

Comment:
unsure about this - maybe it's safer to pick one

Reply:
Thinking some more ... this also chooses which configuration is shown by default.  I don't think we'll be able to make a good choice for the user.  So I'm inclined toward `undefined` - don't attempt to do that.